server/manager/src/main/java/org/apache/accumulo/manager/upgrade/Upgrader11to12.java

```
@@ -0,0 +1,208 @@
+      } catch (MutationsRejectedException ex) {
+        // include constraint violation info in log - but stop upgrade
+        log.warn(
+            "Failed to update file reference for table: " + tableName + ". row: " + key.getRow());
+        log.warn("Constraint violations: {}", ex.getConstraintViolationSummaries());
+        throw new IllegalStateException("File conversion failed. Aborting upgrade", ex);
+      }
+    }
+  }
+
+  @VisibleForTesting
+  void removeChoppedCF(final Key key, final BatchWriter batchWriter, final String tableName) {
+    Mutation delete = null;
+    try {
+      delete = new Mutation(key.getRow()).at().family(ChoppedColumnFamily.STR_NAME)
+          .qualifier(ChoppedColumnFamily.STR_NAME).delete();
+      log.warn(
+          "Deleting chopped reference from:{}. Previous split or delete may not have completed cleanly. Ref: {}",
+          tableName, delete.prettyPrint());
+      batchWriter.addMutation(delete);
+    } catch (MutationsRejectedException ex) {
+      log.warn("Failed to delete obsolete chopped CF reference for table: " + tableName + ". Ref: "
+          + delete.prettyPrint() + ". Will try to continue. Ref may need to be manually removed");
+      log.warn("Constraint violations: {}", ex.getConstraintViolationSummaries());
+    }
+  }
+
+  @VisibleForTesting
+  void removeExternalCompactionCF(final Key key, final BatchWriter batchWriter,
+      final String tableName) {
+    Mutation delete = null;
+    try {
+      delete = new Mutation(key.getRow()).at().family(ExternalCompactionColumnFamily.NAME)
+          .qualifier(key.getColumnQualifier()).delete();
+      log.warn(
+          "Deleting external compaction reference from:{}. Previous compaction may not have completed. Ref: {}",
+          tableName, delete.prettyPrint());
+      batchWriter.addMutation(delete);
+    } catch (MutationsRejectedException ex) {
+      log.warn("Failed to delete obsolete external compaction CF reference for table: " + tableName
```

Also need to throw an exception here.

---

Added in bceac27d1c